### PR TITLE
Clean up dev dependencies

### PR DESCRIPTION
**Pull request by Pessimistress**
See inline comments

#### Change List
- Remove unused dev dependencies from root package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,25 +47,16 @@
     "jsdom": false
   },
   "devDependencies": {
-    "@arcgis/core": "^4.21.0",
```
Comment (Pessimistress) [inline package.json:50]: modules/arcgis specific dependency, moved to submodule

File: package.json [hunk 2]
```diff
@@ -47,25 +47,16 @@
...
-    "@babel/register": "^7.13.0",
```
Comment (Pessimistress) [inline package.json:51]: Babel no longer used

File: package.json [hunk 3]
```diff
@@ -47,25 +47,16 @@
...
     "@loaders.gl/csv": "^4.2.0-alpha.5",
     "@loaders.gl/polyfills": "^4.2.0-alpha.5",
-    "@luma.gl/test-utils": "^9.0.0",
```
Comment (Pessimistress) [inline package.json:54]: Already a dependency of modules/test-utils

File: package.json [hunk 4]
```diff
@@ -47,25 +47,16 @@
...
-    "@probe.gl/test-utils": "^4.0.9",
-    "@types/react": "^18.2.0",
-    "abortcontroller-polyfill": "^1.5.0",
-    "babel-loader": "^8.0.0",
     "jsdom": "^20.0.0",
     "ocular-dev-tools": "2.0.0-alpha.29",
     "pre-commit": "^1.2.2",
     "pre-push": "^0.1.1",
     "puppeteer": "^22.4.0",
-    "react": "^18.2.0",
```
Comment (Pessimistress) [inline package.json:67]: modules/react specific dependency, moved to submodule

File: package.json [hunk 5]
```diff
@@ -47,25 +47,16 @@
...
-    "@babel/register": "^7.13.0",
     "@loaders.gl/csv": "^4.2.0-alpha.5",
     "@loaders.gl/polyfills": "^4.2.0-alpha.5",
-    "@luma.gl/test-utils": "^9.0.0",
     "@luma.gl/webgpu": "^9.0.0",
     "@math.gl/proj4": "^4.0.0",
     "@probe.gl/bench": "^4.0.9",
-    "@probe.gl/test-utils": "^4.0.9",
-    "@types/react": "^18.2.0",
-    "abortcontroller-polyfill": "^1.5.0",
```
Comment (Pessimistress) [inline package.json:60]: Not used

**Review by felixpalmer — APPROVED** [review]

**Pull request merged**